helix-term/src/ui/picker.rs

```diff
@@ -88,13 +90,16 @@ impl<T> FilePicker<T> {
         format_fn: impl Fn(&T) -> Cow<str> + 'static,
         callback_fn: impl Fn(&mut Editor, &T, Action) + 'static,
         preview_fn: impl Fn(&Editor, &T) -> Option<FileLocation> + 'static,
-    ) -> Self {
-        Self {
-            picker: Picker::new(false, options, format_fn, callback_fn),
-            truncate_start: true,
-            preview_cache: HashMap::new(),
-            read_buffer: Vec::with_capacity(1024),
-            file_fn: Box::new(preview_fn),
+    ) -> Overlay<Self> {
+        Overlay {
+            content: Self {
+                picker: Picker::new(options, format_fn, callback_fn),
+                truncate_start: true,
+                preview_cache: HashMap::new(),
+                read_buffer: Vec::with_capacity(1024),
+                file_fn: Box::new(preview_fn),
+            },
+            calc_child_size: Box::new(|rect: Rect| clip_rect_relative(rect.clip_bottom(2), 90, 90)),
         }
     }
```

Comment:
Let's return `FilePicker` here and have it wrapped in an overlay inside the `ui::file_picker` method. That makes it reusable in cases where we don't want to render it centered but instead nest it in a Popup for example.